exe_2.py

```diff
@@ -0,0 +1,67 @@
+class Node(object):
+	def __init__(self, data=None, next_node=None):
+		self.data = data
+		self.next_node = next_node
+
+	def get_data(self):
+		return self.data
+
+	def get_next(self):
+		return self.next_node
+
+	def set_next(self, new_next):
+		self.next_node = new_next
+
+class LinkedList(object):
+	def __init__(self, head=None):
+		self.head = head
+
+	def insert(self, data):
+		new_node = Node(data)
+		new_node.set_next(self.head)
+		self.head = new_node
+
+	def size(self):
+		current = self.head
+		count = 0
+		while current:
+			count += 1
+			current = current.get_next()
+		return count
+
+	def print_(self):
+		current = self.head
+		while current:
+			print str(current.get_data())
+			current = current.get_next()
+
+
+def print_kth_last(linked_list, k):
+	size = linked_list.size()
+	current = linked_list.head
+	linked_list.print_()
+	count = 0
+	if k < size and k >= 0:
+		while count < size-k-1:
+			count += 1
+			current = current.get_next()
+		print str(k), str(current.get_data())
+	else:
+		print 'enter valid k'
+
+def test_code():
+	node1 = Node(1)
+	node2 = Node(2)
+	node3 = Node(3)
+	node4 = Node(4)
+	node5 = Node(5)
+	node1.set_next(node2)
+	node2.set_next(node3)
+	node3.set_next(node4)
+	node4.set_next(node5)
+
+	linked_list = LinkedList()
+	linked_list.head = node1
+	print_kth_last(linked_list, 1)
+
+test_code()
```

Comment:
The complexity is currently O(n), and the algorithm is traversing in the worst case the linked list twice. This algorithm cannot go better than O(n), however we could avoid traversing twice by using two pointers, where the first one would move k nodes before the second one. This way when the end is reached by first pointer, the second pointer would be pointing to the k-th to last element in the list.
Another way would be to change the data structure so that it holds a pointer to the last element. This way it would be just finding the k-th element from tail, which is O(k) runtime, but still linear runtime and O(n) in worst case. Thank you for the question :)